import_tracker/rest.py

```diff
@@ -76,13 +81,18 @@ def getImports(query=None, user=None, unique=False, limit=None, offset=None, sor
     return processCursor(cursor, user)
 
 
-def moveLeafFiles(folder, user, assetstore):
+def moveLeafFiles(folder, user, assetstore, progress, job):
+    if job['status'] == JobStatus.CANCELED:
```

Comment:
You'll need to reload the job from the database to check if the status changed (e.g., `job = Job().load(job['_id'], force=True)`, since otherwise we'll never see the cancel message.